src/core/datastore/deltaset.js

```diff
@@ -0,0 +1,99 @@
+import isArray from 'lodash/isArray';
+import { format } from 'url';
+import { Query } from '../query';
+import { KinveyRequest, RequestMethod, AuthType } from '../request';
+import { Client } from '../client';
+import { repositoryProvider } from './repositories';
+import { stripTagFromCollectionName, generateEntityId } from './utils';
+
+const QUERY_CACHE_COLLECTION_NAME = '_QueryCache';
+
+export function deltaSet(collectionNameWithTag, query, options) {
+  const collectionName = stripTagFromCollectionName(collectionNameWithTag);
+  const serializedQuery = query ? JSON.stringify(query.toQueryString()) : '';
+
+  return repositoryProvider.getOfflineRepository()
+    .then((offlineRepo) => {
+      const queryCacheQuery = new Query()
+        .equalTo('collectionName', collectionName)
+        .and()
+        .equalTo('query', serializedQuery);
+      return offlineRepo.read(QUERY_CACHE_COLLECTION_NAME, queryCacheQuery)
+        .then((deltaSetQueryDocs = []) => {
+          if (deltaSetQueryDocs.length > 0) {
+            return deltaSetQueryDocs[0];
+          }
+
+          return {
+            _id: generateEntityId(),
+            collectionName: collectionName,
+            query: serializedQuery
+          };
+        })
+        .then((deltaSetQueryDoc) => {
+          const client = Client.sharedInstance();
+          let url = format({
+            protocol: client.apiProtocol,
+            host: client.apiHost,
+            pathname: `/appdata/${client.appKey}/${collectionName}`
+          });
+
+          if (deltaSetQueryDoc.lastRequest) {
+            url = format({
+              protocol: client.apiProtocol,
+              host: client.apiHost,
+              pathname: `/appdata/${client.appKey}/${collectionName}/_deltaset`,
+              query: { since: deltaSetQueryDoc.lastRequest }
+            });
+          }
+
+          const request = new KinveyRequest({
+            authType: AuthType.Session,
```

Comment:
This shouldn't be allowed when using the SDK with master secret auth?

Reply:
I changed it to use `AuthType.Default`.